DeviceBridgeE2E/EchoAzureFunction/DeviceBridgeE2EEcho/DeviceBridgeE2EEcho.cs

```diff
@@ -0,0 +1,54 @@
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace FunctionApp1
+{
+    public static class DeviceBridgeE2EEcho
+    {
+        static Dictionary<string, string> cache = new Dictionary<string, string>();
```

Comment:
Can you switch to a `ConcurrentDictionary`? (IIRC az functions will spin up a new thread per request, so we might see some weird behaviors)